src/receive_imf.rs

```diff
@@ -245,12 +245,16 @@ pub(crate) async fn receive_imf_inner(
     }
 
     if let Some(ref status_update) = mime_parser.webxdc_status_update {
-        if let Err(err) = context
-            .receive_status_update(from_id, insert_msg_id, status_update)
-            .await
-        {
-            warn!(context, "receive_imf cannot update status: {}", err);
-        }
+        // protect against users which falsely sent status updates because they
+        // didn't yet receive the group-leave message
+        if is_contact_in_chat(context, chat_id, from_id).await? {
```

Comment:
You will have to check whether from_id is ContactId::SELF, too, because for 1:1 chats ContactId::SELF is not added to the chats_contacts table. Maybe this fixes the failing tests.

Reply:
That didn't solve it. It is somehow dependend on that the updates are marked to be trashed

Reply:
Ah wait, that actually only fixed one of the tests - the test I was working with